bootstrap/config/config.go

```diff
@@ -550,3 +555,29 @@ func (cp *Processor) listenForChanges(serviceConfig interfaces.Configuration, co
 func (cp *Processor) logConfigInfo(message string, overrideCount int) {
 	cp.lc.Infof("%s (%d envVars overrides applied)", message, overrideCount)
 }
+
+// getSecretPathsChanged returns an array of changed secrets and paths between two maps.
+func getSecretPathsChanged(m1 config.InsecureSecrets, m2 config.InsecureSecrets) []string {
+	var updatedPaths []string
+	for key1, val1 := range m1 {
+		if val1.Path == m2[key1].Path {
+			for k, v := range val1.Secrets {
+				// Checking for changed secret.
+				if m2[key1].Secrets[k] != v {
+					updatedPaths = append(updatedPaths, val1.Path)
+				}
+			}
+		} else {
+			// Handling for changes to a Path and disallow empty paths.
+			if len(val1.Path) > 0 {
+				updatedPaths = append(updatedPaths, val1.Path)
+			}
+
+			if len(m2[key1].Path) > 0 {
+				updatedPaths = append(updatedPaths, m2[key1].Path)
+			}
+		}
+	}
+
+	return updatedPaths
+}
```

Comment:
This need to be refactored to do deep compare of entries in the top level map if neither are nil. Something like this

```suggestion
func getSecretPathsChanged(prevVals config.InsecureSecrets, curVals config.InsecureSecrets) []string {
	var updatedPaths []string
	
       for key, prevVal := range prevVals{
          curVal:= curVals[key]
          // Catches removed secrets
          if curVal== nil {
             updatedPaths = append(updatedPaths, prevVal.Path)
             continue
          }

         // Catches changes to secret data or to the path name
         if !reflect.DeepEqual(prevVal,curVal]) {
              updatedPaths = append(updatedPaths, curVal.Path)

             // Catches path name changes, so also include the previous path
             if prevVal.path != curVal.path {
                     updatedPaths = append(updatedPaths, prevVal.path)
             }
         }
      }
      
      for key, curVal := range curVals{
         // Catches new secrets added
         if prevVals[key] == nil {
            updatedPaths = append(updatedPaths, curVal.Path)
          }
       }

	return updatedPaths
}
```